content/en/blog/improving-observability-opentelemetry-datadog-fission/index.md

```diff
@@ -0,0 +1,288 @@
+categories = ["Tutorial","Application","Python","Observability"]
+type = "blog"
++++
+
+{{< imgproc opentelemetry-datadog-fission-featured.png Fit "1000X1000" >}}
+Integrating OpenTelemetry & Datadog with Fission functions
+{{< /imgproc >}}
+
+Software development isn’t just about writing a piece of code and running it.
+There are a lot of other processes around it that ensure that your code is going to work as expected in the real world.
+Amongst all such practices, **Observability** is going to be our point of discussion today.
+
+Observability is all about gaining greater control and visibility over your application.
+It actually tells you what is really happening in your application.
+With most of the applications today adopting a microservices-oriented architecture, the applications have become complex with a lot of moving parts.
+And hence observability is a key to build a robust and reliable system.
+**Logs**, **Metrics** and **Trace** are the **3 pillars of observability**. While logs and metrics are very helpful, trace allows you to traverse the entire journey of a request.
+
+There are a handful of applications out there that help you with observability.
+However, one of the major problems with them is that they aren’t vendor agnostic.
+It means that if you want to change your observability tool, you will have to rewrite the code as well.
+That's where **OpenTelemetry** comes into the picture.
+
+## OpenTelemetry & Datadog
+
+OpenTelemetry gives us vendor-neutral APIs that helps us collect and export our traces to any 3rd party application.
+That means that OpenTelemetry will help you collect the traces and send it to any application that you want to by just changing the configuration.
+You can export your traces to applications like Prometheus, Jaeger, Datadog or anything else that you use.
+
+**Datadog** on the other hand is an observability service for cloud applications.
+It allows you to monitor databases, servers, applications, tools & services.
+It provides you with multiple integration options from where we could send metrics, logs and trace data to Datadog.
+OpenTelemetry is one of the integrations that we can use to share traces from our application to Datadog.
+
+## Integrating OpenTelemetry & Datadog with Fission for enhanced Tracing
+
+Now that we have a basic understanding of observability, OpenTelemetry and Datadog, let’s try to integrate all of them!
+In today’s blog post, I’ll show you how we can integrate OpenTelemetry and Datadog with Fission functions for enhanced tracing.
+This example will show you how you can create custom traces in your Fission functions, and then using OpenTelemetry send them to Datadog.
+The sample code is written in **Python** and uses **OpenTelemetry SDK** along with **Datadog** and **Fission**
+
+{{< imgproc architecture.png Fit "1000X1000" >}}
+Integrating OpenTelemetry & Datadog with Fission functions
```

Comment:
Please avoid using the same text description for images.

Reply:
Missed updating it, fixed it.